### PR TITLE
NONEVM-745 LogPoller db models

### PR DESCRIPTION
**Pull request by dhaidashenko**
Depends on:
* [DB schema in chainlink](https://github.com/smartcontractkit/chainlink/pull/15171)
* [Utils function to use TX DB](https://github.com/smartcontractkit/chainlink-common/pull/910)

### CODE REVIEW

File: pkg/solana/logpoller/models.go
Comment (reductionista) [inline pkg/solana/logpoller/models.go]: Why not `time.Duration`? That's what we use for the Retention field of `logpoller.Filter` on evm

**Comment by reductionista** [comment]
I was thinking for this JIRA we would add a definition for `LogPollerBlock` and  the `LogPoller` interface as well.  I guess those would go in a separate file than `models.go` though (`logpoller.go` or maybe `types.go`?).  Type aliases for `solana.PublicKey`, `solana.Signature`, and `solana.Hash` should go here though.

**Comment by reductionista** [comment]
Wow! It looks like you did a lot more work here than was necessary at present. That's great to see, although I'm a bit confused as to why this is all in this PR as opposed to separate PR's.

It says NONEVM-745 in the title, but it looks like a combination of that, NONEVM-746, and some of NONEVM-744:

https://smartcontract-it.atlassian.net/browse/NONEVM-744
https://smartcontract-it.atlassian.net/browse/NONEVM-745
https://smartcontract-it.atlassian.net/browse/NONEVM-746

NONEVM-746 (Implementation of the ORM) has been blocked on the db prod + unit testing PR's I'm working on. But it looks like you've managed to work around that for now?

File: pkg/solana/logpoller/orm_test.go
Comment (reductionista) [inline pkg/solana/logpoller/orm_test.go]: I'm really confused... I have a PR open to move `NewSqlDB` and some other things over from `chainlink` to `chainlink-common`, but as far as I know it hasn't been merged yet. 🤔

I wonder if someone else moved it over already.

**Comment by reductionista** [comment]
Could you add links in the description here and in https://github.com/smartcontractkit/chainlink/pull/15171 to each other and any other PR's they depend on?

File: pkg/solana/logpoller/orm_test.go
Comment (dhaidashenko) [reply pkg/solana/logpoller/orm_test.go]: Yes, it's not merged. I've used your [PR](https://github.com/smartcontractkit/chainlink-common/pull/910) as a dependency to unblock myself.

**Comment by dhaidashenko** [comment]
> Wow! It looks like you did a lot more work here than was necessary at present. That's great to see, although I'm a bit confused as to why this is all in this PR as opposed to separate PR's.
> 
> It says NONEVM-745 in the title, but it looks like a combination of that, NONEVM-746, and some of NONEVM-744:
> 
> https://smartcontract-it.atlassian.net/browse/NONEVM-744 https://smartcontract-it.atlassian.net/browse/NONEVM-745 https://smartcontract-it.atlassian.net/browse/NONEVM-746
> 
> NONEVM-746 (Implementation of the ORM) has been blocked on the db prod + unit testing PR's I'm working on. But it looks like you've managed to work around that for now?

Yeah, I had to do some additional work that exceeds the scope of the NONEVM-745 to ensure that we can actually read/write into defined models from the database. 
But this PR does not fully address other tickets that you've referenced. I'm not sure if it makes sense to split this work into more granular ones, given it's still a manageable size and that we are on a tight schedule.

File: pkg/solana/logpoller/query.go
Comment (reductionista) [inline pkg/solana/logpoller/query.go:8]: While reviewing @EasterTheBunny 's DSL Parsing PR, I noticed we're missing unit tests for the functions in this file.  In evm we have a `Test_QueryArgs` set of tests that runs through these in a `query_test.go` file. Should we add something similar here?

Comment (dhaidashenko) [reply pkg/solana/logpoller/query.go:8]: Replied in the DSL Parsing PR https://github.com/smartcontractkit/chainlink-solana/pull/959/files#r1876556494

**Review by reductionista — DISMISSED** [review]

**Review by DylanTinianov — DISMISSED** [review]

File: pkg/solana/logpoller/types.go
Comment (EasterTheBunny) [inline pkg/solana/logpoller/types.go:25]: I don't understand the need for the `ToSolana` methods. They don't follow a common interface since they all return something different so we don't really gain anything on using them. Casting, `solana.PublicKey(value)`, would do the same trick in every case given all of the structs are exported from the package.

File: pkg/solana/logpoller/types.go
Comment (EasterTheBunny) [inline pkg/solana/logpoller/types.go:62]: Frequent errors like this are good to make as exported errors and wrap them whenever possible.

```
var ErrScanFailure = errors.New("failed to scan")

// ...

return fmt.Errorf("%w %T into %s", ErrScanFailure, src, name)
```

This allows calling functions from external packages to optionally handle errors more gracefully.

File: pkg/solana/logpoller/orm_test.go
Comment (jadepark-dev) [inline pkg/solana/logpoller/orm_test.go:132]: love it

**Review by jadepark-dev — DISMISSED** [review]
Code looks good but looks like some of early commits are not signed?

File: pkg/solana/logpoller/types.go
Comment (dhaidashenko) [reply pkg/solana/logpoller/types.go:25]: They are needed to ensure that users of the PublicKey type won't introduce bugs by casting to the wrong type.
The compiler won't show an error if user tries to cast `PublicKey` to  `solana.Hash`

File: pkg/solana/logpoller/types.go
Comment (dhaidashenko) [reply pkg/solana/logpoller/types.go:62]: I am not sure that it makes sense to handle gracefully failure to unmarshal the db row. 
Even if such a use case exists, no one currently uses it. I prefer to avoid introducing public variables that are only used internally.

**Review by DylanTinianov — DISMISSED** [review]

**Review by reductionista — DISMISSED** [review]

**Review by jadepark-dev — APPROVED** [review]

**Pull request merged**
